### PR TITLE
Change Rules In `.eslintrc.js` + Bump `eslint` Version

### PR DESCRIPTION
**Pull request by shirady**
### Explain the changes
1. Add rule `prefer-const` (for more information, see [here](https://eslint.org/docs/latest/rules/prefer-const)) and turn on `no-var` rule in turn off `logical-assignment-operators` rule `.eslintrc.js`.
2. Change files according to the changes in the rules above.
3. Bump `eslint` version to `8.38.0` (change file `package.json` and automatically was generated changes in file `package-lock.json` after `npm install`).

### Issues: Fixed #xxx / Gap #xxx
1. none

### Testing Instructions:
1. none

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

**Comment by shirady** [comment]
As you can see I chose in the rule:
```
        "prefer-const": ["error", {
            "destructuring": "all",
            "ignoreReadBeforeAssign": false
        }],
```
to use  `"destructuring": "all"` and not `"destructuring": "any"` (see [here](https://eslint.org/docs/latest/rules/prefer-const#destructuring)) because we have cases in which we have destructor and part of the variables are `const` and part of them `let`, for example:
https://github.com/noobaa/noobaa-core/blob/81307b6f8806be712b0c500877aa81e5ed651823/src/endpoint/s3/s3_rest.js#L307
(`bucket` is `const`, `key` is `let`...).

File: .git-blame-ignore-revs
Comment (dannyzaken) [inline .git-blame-ignore-revs:2]: shouldn't you add it only after merging to master?

**Comment by dannyzaken** [comment]
@shirady great PR! 

Following up on the slack discussion with @liranmauda, let's change the eslint version to the latest (`v8.38.0`) and set `'logical-assignment-operators': 'off'` in `eslintrc`

File: .git-blame-ignore-revs
Comment (shirady) [reply .git-blame-ignore-revs:2]: (@dannyzaken I added numbers):

> 1. what is this revision? 
> 2. Shouldn't you add it only after merging to master?

1. I'm not sure I understand the first question. if you asked about the file `.git-blame-ignore-revs` it is a file that allows us to do changes that will not be reflected in `git blame` option (in Github as well). [here](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view) is the link for more information.
2. I did it in a separate commit, I will do it in a separate PR (after merging this PR).

**Comment by shirady** [comment]
> @shirady great PR!
> 
> Following up on the slack discussion with @liranmauda, let's change the eslint version to the latest (`v8.38.0`) and set `'logical-assignment-operators': 'off'` in `eslintrc`

I added `'logical-assignment-operators': 'off'`.
@dannyzaken Are we sure that we want to bump the version in this PR? I asked @liranmauda in #7265 if he can bump it there.

**Comment by dannyzaken** [comment]
> > @shirady great PR!
> > Following up on the slack discussion with @liranmauda, let's change the eslint version to the latest (`v8.38.0`) and set `'logical-assignment-operators': 'off'` in `eslintrc`
> 
> I added `'logical-assignment-operators': 'off'`. @dannyzaken Are we sure that we want to bump the version in this PR? I asked @liranmauda in #7265 if he can bump it there.

@liranmauda PR is intended to be backported to 5.13, but we don't want to backport the eslint changes. Also, I think it's relevant to this PR

**Comment by dannyzaken** [comment]
LGTM. @liranmauda waiting for your approval

**Review by liranmauda — APPROVED** [review]

**Pull request merged**
